lgc/patch/PatchInOutImportExport.h

```diff
@@ -222,9 +225,10 @@ class PatchInOutImportExport : public Patch {
   llvm::GlobalVariable *m_lds; // Global variable to model LDS
   llvm::Value *m_threadId;     // Thread ID
 
-  std::vector<llvm::CallInst *> m_importCalls;                 // List of "call" instructions to import inputs
-  std::vector<llvm::CallInst *> m_exportCalls;                 // List of "call" instructions to export outputs
-  PipelineState *m_pipelineState = nullptr;                    // Pipeline state from PipelineStateWrapper pass
+  std::vector<llvm::CallInst *> m_importCalls;          // List of "call" instructions to import inputs
+  std::vector<llvm::CallInst *> m_exportCalls;          // List of "call" instructions to export outputs
+  std::map<unsigned, llvm::Value *[4]> m_attribExports; // Export info of vertex attributes
```

Comment:
Use `std::array` or `SmallVector` for the map value type.

Reply:
As I explained above, here we have a fixed-length array without any push_back operations. All components must be explicitly specified as a complete vertex attribute export.